qiita_pet/handlers/analysis_handlers.py

```diff
@@ -321,3 +325,31 @@ def get(self):
                     user.shared_analyses + user.private_analyses]
 
         self.render("show_analyses.html", analyses=analyses)
+
+
+class ResultsHandler(StaticFileHandler, BaseHandler):
+    def validate_absolute_path(self, root, absolute_path):
+        """Overrides StaticFileHandler's method to include authentication
+        """
+        u = self.current_user
+
+        accessible_filepaths = get_accessible_filepath_ids(u)
+        accessible_filepaths = {join(get_db_files_base_dir(),
+                                     filepath_id_to_rel_path(fpid))
+                                for fpid in accessible_filepaths}
+
+        for accessible_filepath in accessible_filepaths:
+            access = False
+
+            if (accessible_filepath == absolute_path):
+                access = True
+
+            elif isdir(accessible_filepath) and \
+                    absolute_path.startswith(accessible_filepath + sep):
+                access = True
+
+            if access:
+                return super(ResultsHandler, self).validate_absolute_path(
+                    root, absolute_path)
+
+        raise QiitaPetAuthorizationError(u.id, absolute_path)
```

Comment:
From a probing perspective, this might help reveal details about valid paths (despite access) and could expose more details than necessary. Totally minor... but we could also just throw either 404 or just 403 (regardless of if the path exists or not)